main.js

```diff
@@ -294,6 +321,25 @@ export class IdDecoder {
         bitLength: this.fixedBitLength
       });
     }
+    if(this.multihash) {
+      // <varint hash fn code>: identity function
+      const [hashFnCode] = decoded.slice(0, 1);
+      if(hashFnCode !== MULTIHASH_IDENTITY_FUNCTION_CODE) {
+        throw new Error('Invalid multihash function code.');
+      }
+      // <varint digest size in bytes>: 32
+      const [digestSize] = decoded.slice(1, 2);
+
+      if(digestSize !== this.expectedSize) {
+        throw new Error('Invalid digest size.');
+      }
+      const keySeedBytes = decoded.slice(2, decoded.length);
+      if(keySeedBytes.byteLength !== KEY_SEED_BYTE_SIZE) {
```

Comment:
This doesn't look right -- it should be the expected size, right? Also, again this isn't specific to "seeds" -- so that terminology should be removed. This is just about allowing encoded values to include their size as a sanity check.

Reply:
Updated 155dcae